api/v1beta1/azuremachine_default.go

```diff
@@ -174,11 +174,11 @@ func (s *AzureMachineSpec) SetNetworkInterfacesDefaults() {
 	}
 }
 
-// GetSubscriptionID returns the subscription ID for the given cluster name and namespace.
-func GetSubscriptionID(cli client.Client, clusterName string, namespace string, maxAttempts int) (string, error) {
+// GetOwnerAzureClusterNameAndNamespace returns the owner azure cluster's name and namespace for the given cluster name and namespace.
+func GetOwnerAzureClusterNameAndNamespace(cli client.Client, clusterName string, namespace string, maxAttempts int) (azureClusterName string, azureClusterNamespace string, err error) {
 	ctx := context.Background()
 
-	ownerCluster := &AzureCluster{}
+	ownerCluster := &clusterv1.Cluster{}
```

Comment:
Since this "get CAPI Cluster" step is new, is the retry logic here still applicable the same way as for the "get CAPZ AzureCluster" step? Or is there an opportunity to make it more strict or even eliminate it entirely here?

Reply:
Looks like the retries are specific to machinepools from this comment. I will remove the retry logic for now from these util functions.

https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/3080#discussion_r1089303163